components/server/src/workspace/workspace-starter.ts

```diff
@@ -426,14 +433,40 @@ export class WorkspaceStarter {
         reason: string,
         policy?: StopWorkspacePolicy,
     ): Promise<void> {
-        ctx.span?.setTag("stopWorkspaceReason", reason);
+        const span = TraceContext.startSpan("stopWorkspaceInstance", ctx);
+        span.setTag("stopWorkspaceReason", reason);
         log.info({ instanceId }, "Stopping workspace instance", { reason });
 
         const req = new StopWorkspaceRequest();
         req.setId(instanceId);
         req.setPolicy(policy || StopWorkspacePolicy.NORMALLY);
 
-        const client = await this.clientProvider.get(instanceRegion);
+        let client: PromisifiedWorkspaceManagerClient | undefined;
+        try {
+            client = await this.clientProvider.get(instanceRegion);
+        } catch (err) {
```

Comment:
It's too late now, but honestly I'm not super happy with relying on the error here.
Looks like we never invalidate the [connectionCache](https://github.com/gitpod-io/gitpod/blob/0e2ec6cd247e68b45451ee3f0556ef7d20703dcd/components/ws-manager-api/typescript/src/client-provider.ts#L107), for instance, so it's "hit-or-miss" whether we actually are able to remove sth.

Why not use getAllWorkspaceClusters, or add another method for uncached lookup?